titus_isolate/api/status.py

```diff
@@ -111,37 +106,11 @@ def get_wm_status():
     })
 
 
-def health_check(local_exit_handler: ExitHandler, local_event_manager: EventManager, config_manager: ConfigManager):
-    last_processed_event_time_epoch_s = datetime.datetime.utcnow().timestamp()
-
-    while True:
-        try:
-            threshold = config_manager.get_int(
-                MAX_TIME_SINCE_LAST_SUCCESSFUL_EVENT_KEY,
-                DEFAULT_MAX_TIME_SINCE_LAST_SUCCESSFUL_EVENT)
-
-            health_check_interval = config_manager.get_int(
-                HEALTH_CHECK_FREQUENCY_KEY,
-                DEFAULT_HEALTH_CHECK_FREQUENCY)
-
-            last_event_epoch_s = local_event_manager.last_successful_event_epoch_s
-            if last_event_epoch_s > 0:
-                last_processed_event_time_epoch_s = last_event_epoch_s
-
-            time_since_last_event = datetime.datetime.utcnow().timestamp() - last_processed_event_time_epoch_s
-            log.info("Seconds since last successful event processing: %s, threshold: %s",
-                     time_since_last_event, threshold)
-
-            if time_since_last_event > threshold:
-                log.info("Event processing is not completing as expected, exiting...")
-                local_exit_handler.exit(HEALTH_CHECK_FAILURE_EXIT_CODE)
-
-            log.info("Next health check in %s seconds", health_check_interval)
-            time.sleep(health_check_interval)
-
-        except:
-            log.exception("Failed to healthcheck")
-            local_exit_handler.exit(HEALTH_CHECK_EXCEPTION_EXIT_CODE)
+def _notify_ready():
```

Comment:
I think you can look at the environment variable "NOTIFY_SOCKET" here instead of `is_running_on_agent` to see if it's running as a notify unit

Reply:
> Did you already bring the systemd dep in somewhere? If this is a new dep, one alternative might be to use systemd watchdog, which will run a script periodically to check your workload is alive: https://stash.corp.netflix.com/projects/TN/repos/titus-system-tools/browse/systemd-watchdog-wrapper/main.go

Not sure what you mean.  The `systemd-watchdog-wrapper` is needed?  NPD takes care of this now right?

Reply:
systemd-watchdog-wrapper isn't needed, but it's a "lightweight" replacement for the code you have here.